### ISSUE TITLE
How can I train the network for another language?

### ISSUE THREAD

**Issue by s1mplelul** [MichalBusta/DeepTextSpotter#32]
Hi,
I just don't know how to start with the text recognition for another language.
I have tried changing 'codec' variable (add more non-English characters), and I think I should change 'caffe.SpatialConvolution_36' in 'model_cz.prototxt ' with another 'num_output' value. But I don't know how to choose that value.

Could you please give me some direction? 
Thank you very much :)

**Comment by MichalBusta** [comment]
You just need |A| + 1 filters on final conv. layer. Where |A| is number of symbols and 1 if for padding. 

this might help: https://arxiv.org/abs/1801.09919

All the best.

**Comment by s1mplelul** [comment]
Many thanks for the prompt reply.
I just wonder why you use the number 141 instead of 126 (as the idea 125 characters in codec + 1).

Thank you :)

**Comment by MichalBusta** [comment]
On 04/17/2018 06:02 PM, s1mplelul wrote:
>
> Many thanks for the prompt reply.
> I just wonder why you use the number 141 instead of 126 (as the idea 
> 125 characters in codec + 1).
>
I'm just lazy - different data for some demo app - bigger vocab - some 
filters are un-used for ICDAR.

**Comment by s1mplelul** [comment]
Thanks a lot :D I have the answer.

**Comment by s1mplelul** [comment]
Hi, I followed your direction and trained the network on my own dataset. Everything is good until I test the model
the label is always equal [0, 0, 0, ..., 0]
I don't know why :( Did I miss something?

**Comment by MichalBusta** [comment]
Probably yes - it is not that hard, just think :) 
0 is most probable label in any sequence, so your network is in "early state of training"

Cues: 
 - did you pre-train OCR network 
 - what is your validation accuracy
Michal

**Comment by nvtiep** [comment]
I wonder that, the training code is incomplete?
I tried to prepare training and validation data. Here is the format of the ground truth:

`1 0.370434036255 0.900643147213 0.442704864485265 0.0868962200418034 -0.015816627739172726 Reply-To:
1 0.751672108968 0.509530331607 0.04425845094241941 0.018589249706943537 0.5520744494537184 the
1 0.731317240397 0.543194300692 0.04234948363016886 0.017701289179520224 0.4671398519696018 get`

Then I modified some lines of your hard code. For example:
**train.py:**
`parser.add_argument('-data_dir', default='/mnt/textspotter/tmp/SynthText')
parser.add_argument('-train_list', default='/home/busta/data/test_icdar181.txt')
parser.add_argument('-valid_list', default='/home/busta/data/ocr/list.txt')`

**validation.py**
`sys.path.insert(0, '/home/busta/git/caffe/python')`
`cmp_trie.load_dict('/home/busta/data/icdar2013-Test/GenericVocabulary.txt')`

I also found that in train.py there is a variables "labels_gt" used without initializing values (line 104) and "continue" (line 232) without doing something :-)

**Comment by MichalBusta** [comment]
"continue" (line 232) without doing something :-) - sorry :)
